### ISSUE TITLE
rbenv.installed fails when rbenv installed globally

### ISSUE THREAD

**Issue by xuhcc** [saltstack/salt#44811]
### Description of Issue/Question

rbenv is installed globally on the target machine (Arch Linux):

```
$ which rbenv
/usr/bin/rbenv
```

`_rbenv_exec` function in `salt/modules/rbenv.py` returns `False` when salt couldn't find rbenv installation, and this results in error:

```
----------
          ID: myuser_ruby
    Function: rbenv.installed
        Name: 2.3.3
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/var/tmp/.myuser_684964_salt/py2/salt/state.py", line 1843, in call
                  **cdata['kwargs'])
                File "/var/tmp/.myuser_684964_salt/py2/salt/loader.py", line 1795, in wrapper
                  return f(*args, **kwargs)
                File "/var/tmp/.myuser_684964_salt/py2/salt/states/rbenv.py", line 143, in installed
                  return _check_and_install_ruby(ret, name, default, user=user)
                File "/var/tmp/.myuser_684964_salt/py2/salt/states/rbenv.py", line 93, in _check_and_install_ruby
                  if __salt__['rbenv.install_ruby'](ruby, runas=user):
                File "/var/tmp/.myuser_684964_salt/py2/salt/modules/rbenv.py", line 248, in install_ruby
                  if ret['retcode'] == 0:
              TypeError: 'bool' object has no attribute '__getitem__'
     Started: 13:47:50.485521
    Duration: 3677.38 ms
     Changes:   
```

### Setup

```
myuser_ruby:
  rbenv.installed:
    - name: 2.3.3
    - user: myuser
    - require:
      - pkg: ruby_build_package
```

### Versions Report

```
$ salt-ssh --versions
Salt Version:
           Salt: 2017.7.2
 
Dependency Versions:
           cffi: 1.11.2
       cherrypy: Not Installed
       dateutil: 2.6.1
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.10
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.27.0
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: 2.18
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.14 (default, Sep 20 2017, 01:25:59)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.3
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.2
            ZMQ: 4.2.2
 
System Versions:
           dist:   
         locale: UTF-8
        machine: x86_64
        release: 4.13.12-1-ARCH
         system: Linux
        version: Not Installed
 ```

**Comment by gtmanfred** [comment]
Thanks for reporting this, would you mind submitting a PR to fix it?

Otherwise, I have marked this as a bug.

Thanks,
Daniel

**Comment by xuhcc** [comment]
@gtmanfred 

I have submitted simple fix for TypeError (#44900), but the underlying rbenv installation issue is more complicated.
